config-check/src/main/java/com/okta/commons/configcheck/ConfigurationValidator.java

```diff
@@ -42,75 +42,128 @@ private ConfigurationValidator() {}
+     */
+    public static void assertIssuer(String url) {
+        validateIssuer(url).ifInvalidThrow();
+    }
+
+    public static ValidationResponse validateIssuer(String url) {
+        return validateHttpsUrl(url, "issuerUrl");
     }
 
     /**
      * Validates that a client Id is not null and contains does not contain the string {@code {apiToken}}.
      * @param clientId The Client Id to be validated
      */
-    public static void validateClientId(String clientId) {
-        hasText(clientId, ERRORS.getString("clientId.missing"));
-        doesNotContain(clientId, "{clientId}",  ERRORS.getString("clientId.containsBrackets"));
+    public static void assertClientId(String clientId) {
+        validateClientId(clientId).ifInvalidThrow();
+    }
+
+    public static ValidationResponse validateClientId(String clientId) {
+        ValidationResponse response = new ValidationResponse();
+        if (!hasText(clientId)) {
+            response.setMessage(ERRORS.getString("clientId.missing"));
+        } else if (contains(clientId, "{clientId}")) {
+            response.setMessage(ERRORS.getString("clientId.containsBrackets"));
+        }
+        return response;
     }
 
     /**
      * Validates that a client secret is not null and contains does not contain the string {@code {clientSecret}}.
      * @param clientSecret the Client Secret to be validated
      */
-    public static void validateClientSecret(String clientSecret) {
-        hasText(clientSecret, ERRORS.getString("clientSecret.missing"));
-        doesNotContain(clientSecret, "{clientSecret}",  ERRORS.getString("clientSecret.containsBrackets"));
+    public static void assertClientSecret(String clientSecret) {
+        validateClientSecret(clientSecret).ifInvalidThrow();
+    }
+
+    public static ValidationResponse validateClientSecret(String clientSecret) {
+        ValidationResponse response = new ValidationResponse();
+        if (!hasText(clientSecret)) {
+            response.setMessage(ERRORS.getString("clientSecret.missing"));
+        } else if (contains(clientSecret, "{clientSecret}")) {
+            response.setMessage(ERRORS.getString("clientSecret.containsBrackets"));
+        }
+        return response;
     }
 
     private static String formattedErrorMessage(String messageKey, Object... args) {
         String message = ERRORS.getString(messageKey);
         return MessageFormat.format(message, args);
     }
 
+    private static ValidationResponse validateHttpsUrl(String url, String keyPrefix) {
+        ValidationResponse response = new ValidationResponse();
+        if (!hasText(url)) {
+            response.setMessage(ERRORS.getString(keyPrefix + ".missing"));
+        } else if (contains(url, "{yourOktaDomain}")) {
+            response.setMessage(ERRORS.getString(keyPrefix + ".containsBrackets"));
+        } else {
+            try {
+                URL tempUrl = new URL(url);
+                if (!"https".equalsIgnoreCase(tempUrl.getProtocol())) {
```

Comment:
Will this screw up developers in general, or our own internal developers that might want to test against a locally running okta org? Should we check for an allow local flag or something?

Reply:
possibly, that is a topic of much debate right now.  We will likely have a `okta.testing.disableHttpsCheck` property